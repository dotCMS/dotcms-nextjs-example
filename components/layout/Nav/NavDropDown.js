import { useState } from "react";
import PropTypes from "prop-types";
import ItemLink from "./ItemLink";

const NavDropDown = ({ options }) => {
  const [focus, setFocus] = useState(false);

  const classStatus = `rd-nav-item rd-navbar--has-megamenu rd-navbar-submenu ${
    focus ? "focus" : ""
  }`;

  return (
    <ItemLink
      pathname={options.href}
      className={classStatus}
      dropDown="true"
      topMenu="true"
      onMouseOut={() => {
        setFocus(false);
      }}
    >
      <a
        href={options.href}
        className="rd-nav-link"
        onMouseOver={() => {
          setFocus(true);
        }}
        
      >
        {options.title}
      </a>
      <span className="rd-navbar-submenu-toggle"></span>

      <ul onMouseOver={() => {
          setFocus(true);
        }} className="rd-menu rd-navbar-megamenu rd-navbar-open-left rd-navbar-open-right">
        {options.children.map((subItem, index) => {
          return (
            <li className="rd-megamenu-item" key={index}>
              <div className="rd-megamenu-title">
                <ItemLink pathname={subItem.href} target={subItem.target}>
                  {subItem.title}
                </ItemLink>
              </div>
              <ul className="rd-megamenu-list">
                {subItem.children.map((extraItem, k) => {
                  return (
                    <li className="rd-megamenu-list-item" key={k}>
                      <ItemLink
                        className="rd-dropdown-link"
                        pathname={extraItem.href}
                        target={extraItem.target}
                      >
                        {extraItem.title}
                      </ItemLink>
                    </li>
                  );
                })}
              </ul>
            </li>
          );
        })}
      </ul>
    </ItemLink>
  );
};

NavDropDown.propTypes = {
  options: PropTypes.shape({
    title: PropTypes.string.isRequired,
    children: PropTypes.arrayOf(
      PropTypes.shape({
        folder: PropTypes.string,
        href: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired
      })
    )
  }).isRequired
};

export default NavDropDown;
