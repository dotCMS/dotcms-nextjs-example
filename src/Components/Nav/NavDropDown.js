import React from 'react';
import PropTypes from 'prop-types';
import { DropdownItem, DropdownMenu, DropdownToggle, UncontrolledDropdown } from 'reactstrap';

const NavDropDown = ({ options }) => {
    return (
        <UncontrolledDropdown nav inNavbar>
            <DropdownToggle nav caret>
                {options.title}
            </DropdownToggle>
            <DropdownMenu right>
                {options.children.map((subItem, index) => {
                    return (
                        <DropdownItem
                            tag="a"
                            href={subItem.href}
                            active={subItem.href === window.location.pathname}
                            key={index}
                        >
                            {subItem.title}
                        </DropdownItem>
                    );
                })}
            </DropdownMenu>
        </UncontrolledDropdown>
    );
};

NavDropDown.propTypes = {
    options: PropTypes.shape({
        title: PropTypes.string.isRequired,
        children: PropTypes.arrayOf(
            PropTypes.shape({
                folder: PropTypes.string,
                href: PropTypes.string.isRequired,
                title: PropTypes.string.isRequired
            })
        )
    }).isRequired
};

export default NavDropDown;
