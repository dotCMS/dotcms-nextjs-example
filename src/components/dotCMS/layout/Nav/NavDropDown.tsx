// Dependencies
import * as React from 'react'

// Internals
import { Link } from '@/components'

export type DropdownItem = {
  folder: string
  href: string
  title: string
  children?: DropdownItem[]
}

export type NavDropDownProps = {
  options: DropdownItem
}

export const NavDropDown = ({ options }: NavDropDownProps) => {
  const [focus, setFocus] = React.useState(false)

  return (
    <li
      className={`rd-nav-item rd-navbar--has-megamenu rd-navbar-submenu ${
        focus ? 'focus' : ''
      }`}
      onMouseOut={() => {
        setFocus(false)
      }}
      onMouseOver={() => {
        setFocus(true)
      }}
    >
      <Link className="rd-nav-link" href={options.href}>
        {options.title}
      </Link>
      <span className="rd-navbar-submenu-toggle" />
      <ul className="rd-menu rd-navbar-megamenu rd-navbar-open-left rd-navbar-open-right">
        {options.children?.map((subItem, index) => {
          return (
            <li className="rd-megamenu-item" key={index}>
              <div className="rd-megamenu-title">
                <Link href={subItem.href}>{subItem.title}</Link>
              </div>
              <ul className="rd-megamenu-list">
                {subItem.children?.map((extraItem, k) => {
                  return (
                    <li className="rd-megamenu-list-item" key={k}>
                      <Link className="rd-dropdown-link" href={extraItem.href}>
                        {extraItem.title}
                      </Link>
                    </li>
                  )
                })}
              </ul>
            </li>
          )
        })}
      </ul>
    </li>
  )
}

export default NavDropDown
