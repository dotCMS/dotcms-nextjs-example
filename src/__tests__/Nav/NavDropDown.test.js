import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import NavDropDown from "../../Components/Nav/NavDropDown";

const optionsMock = {
    href: '/resources',
    title: 'Resources',
    folder: '789',
    children: [
        {
            href: '/videos',
            title: 'Videos',
            folder: null
        },
        {
            href: '/blog',
            title: 'Blog',
            folder: null
        }
    ]
};

describe('<NavDropDown />', () => {
    it('renders NavOption correctly', () => {
        const wrapper = shallow(<NavDropDown options={optionsMock}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

});
