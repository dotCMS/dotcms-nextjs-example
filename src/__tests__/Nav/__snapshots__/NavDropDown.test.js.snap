// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavDropDown /> renders NavOption correctly 1`] = `
<UncontrolledDropdown
  inNavbar={true}
  nav={true}
>
  <DropdownToggle
    aria-haspopup={true}
    caret={true}
    color="secondary"
    nav={true}
  >
    Resources
  </DropdownToggle>
  <DropdownMenu
    flip={true}
    right={true}
    tag="div"
  >
    <DropdownItem
      active={false}
      className="dropdown-item"
      href="/videos"
      key="0"
      tag="a"
      toggle={true}
    >
      Videos
    </DropdownItem>
    <DropdownItem
      active={false}
      className="dropdown-item"
      href="/blog"
      key="1"
      tag="a"
      toggle={true}
    >
      Blog
    </DropdownItem>
  </DropdownMenu>
</UncontrolledDropdown>
`;
